File: backend/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookstoreApi.Models;

// Book entity model that maps to the Books table in the SQLite database.
// Each property corresponds to a column in the table.
public class Book
{
    [Key]
    public int BookID { get; set; }             // Primary key
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public string Publisher { get; set; } = string.Empty;
    public string ISBN { get; set; } = string.Empty;
    public string Classification { get; set; } = string.Empty; // Fiction or Non-Fiction
    public string Category { get; set; } = string.Empty;       // e.g. Biography, Self-Help
    public int PageCount { get; set; }
    public double Price { get; set; }
}

File: frontend/src/types/Book.ts
// Represents a single book from the API.
// Each field matches a column in the Books database table.
export interface Book {
  bookID: number;
  title: string;
  author: string;
  publisher: string;
  isbn: string;
  classification: string;
  category: string;
  pageCount: number;
  price: number;
}

// Response shape returned by the paginated GET /api/books endpoint.
// Includes the list of books and pagination metadata.
export interface BookResponse {
  books: Book[];
  totalNumBooks: number;
  pageSize: number;
  currentPage: number;
  totalPages: number;
}

// Represents a book and its quantity in the shopping cart.
export interface CartItem {
  book: Book;
  quantity: number;
}
